py/mpprint.h

```diff
@@ -80,3 +80,59 @@ int mp_vprintf(const mp_print_t *print, const char *fmt, va_list args);
 #endif
 
 #endif // MICROPY_INCLUDED_PY_MPPRINT_H
+
+// You can use several MP_PRN_LEVEL defines in separate _.c files
+#if defined(MP_PRN_LEVEL) && (MP_PRN_LEVEL > 0)
+// Debug messages during code developing with MP_PRN(level, ...) & MP_PRN_LEVEL.
+// An approximate hierarchy of debug levels MP_PRN_LEVEL is:
+#define MP_PRN_SUPPRESS 0 // SUPPRESS all messages. Use it in the release version.
+#define MP_PRN_CRITICAL 1 // For the most CRITICAL errors, often requiring a system reset. Use a message with this level, if possible, raising an exception.
+#define MP_PRN_ERROR 2 // ERROR requiring program restart, use message with this level before raising an exception.
+#define MP_PRN_WARNING 3 // WARNING, something went wrong, but you can fix it with additional operations in code right now or may ignore it.
+#define MP_PRN_INFO 4 // INFO, it is interesting and useful for understanding a bug.
+#define MP_PRN_DEBUG 5 // DEBUG, more detailed information, dig deeper.
+#define MP_PRN_TRACE 6 // TRACE, show a flow of the algorithm, like enter/exit a function.
+// In reality, you may use your own classification of debug levels.
+
+#if defined(MP_PRN)
+#undef MP_PRN
+#endif
+
+#define MP_PRN(level, ...) \
+    do { \
+        if ((0 < level) && (level <= MP_PRN_LEVEL)) { \
+            mp_printf(MP_PYTHON_PRINTER, " MP_PRN_LEVEL=%d : ", level); \
+            mp_printf(MP_PYTHON_PRINTER, __VA_ARGS__); \
+            mp_printf(MP_PYTHON_PRINTER, "\t : FUNC=%s LINE=%d FILE=%s\n", __FUNCTION__, __LINE__, __FILE__); \
```

Comment:
This is a very verbose line to be printing at all levels; on many boards the uart output can be quite slow so more text here can seriously slow down the program

Reply:
My experience with debugging two files showed satisfactory speed. Most of the time I didn't print too much information by lowering MP_DEBUG_PRINT_LEVEL or pushing MP_DEBUG_PRINT to a level above MP_DEBUG_PRINT(MP_DEBUG_PRINT_XXX +1, ...) or two
MP_DEBUG_PRINT(MP_DEBUG_PRINT_XXX +2, ...).
In extreme cases, a developer can correct MP_DEBUG_PRINT as he wishes.

Reply:
This is highly dependent on the particular board in use, and the importance of speed in the code bring tested. The existing debug prints lines in much of the Bluetooth code for instance and very short because it breaks Bluetooth if they're much longer ...